File: Python/Contains Duplicate.py
# import time
class Solution:
    def containsDuplicate(self, nums) -> bool:
        counts = dict()
        for i in nums:
            if i not in counts.keys():
                counts[i] = 1
            else:
                return True
        return False
    
# start = time.time()
s = Solution()
nums = [1,2,3,4]
print(s.containsDuplicate(nums))
# print(f"Total time taken: {time.time() - start}")

File: CPP/Contains_Duplicate.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_map<int, int> counts;
        int n = nums.size();
        for (int i = 0; i < n; i++){
            if (counts.find(nums[i]) != counts.end()){
                return true;
            }
            counts[nums[i]] = 1;
        }
        return false;
    }
};

int main(){
    Solution s;
    vector<int> nums {1,2,3,1};
    cout << s.containsDuplicate(nums);
    return 0;
}